glean-core/src/lib.rs

```diff
@@ -557,8 +557,18 @@ pub fn shutdown() {
         glean.set_dirty_flag(false);
     });
 
-    // We need to wait for above task to finish.
-    dispatcher::block_on_queue();
+    // We need to wait for above task to finish,
+    // but we also don't wait around forever.
+    //
+    // TODO: Make the timeout configurable?
```

Comment:
In uploader_shutdown we have the TODO sorta-kinda satisfiable by looking at the data and determining what the "normal" length of time for uploader shutdown is. Would it be unreasonable to ask for a similar piece of instrumentation and a follow-up bug to look at the data?

Reply:
The wonderful world of instrumentation when we kinda know we're running into trouble.
As long as we do _not_ timeout we should be ok recording this, but when running into a timeout due to I/O in the dispatcher doing more I/O might just as well fail. Nonetheless, yes, we should instrument.